fe/fe-core/src/main/java/com/starrocks/analysis/Delimiter.java

```diff
@@ -31,29 +31,97 @@ public static String convertDelimiter(String originStr) throws AnalysisException
-                }
+        StringWriter writer = new StringWriter();
+
+        for (int i = 0; i < originStr.length(); i++) {
+            char ch = originStr.charAt(i);
+            boolean outputOneChar = true;
+            if (ch == '\\') {
+               char nextChar = (i == originStr.length() - 1) ? '\\' : originStr.charAt(i + 1);
+               switch (nextChar) {
+                   case '\\':
+                       ch = '\\';
+                       break;
+                   case 'b':
+                       ch = '\b';
+                       break;
+                   case 'f':
+                       ch = '\f';
+                       break;
+                   case 'n':
+                       ch = '\n';
+                       break;
+                   case 'r':
+                       ch = '\r';
+                       break;
+                   case 't':
+                       ch = '\t';
+                       break;
+                   case 'x':
+                   case 'X': {
+                       outputOneChar = false;
+                       i = parseHexString(originStr, writer, i);
+                       break;
+                   }
+                   case '\"':
+                       ch = '\"';
+                       break;
+                   case '\'':
+                       ch = '\'';
+                       break;
+                   default:
+                       writer.append(ch);
+                       continue;
+               }
+               if (outputOneChar) {
+                   writer.append(ch);
+                   i++;
+               }
+               // compatible previous 0x / 0X prefix
+            } else if (ch == '0' && i != originStr.length() - 1
+                    && (originStr.charAt(i + 1) == 'x' || originStr.charAt(i + 1) == 'X')) {
+                i = parseHexString(originStr, writer, i);
+            } else {
+                writer.append(ch);
             }
+        }
+
+        return writer.toString();
+    }
 
-            // transform to delimiter
-            StringWriter writer = new StringWriter();
-            for (byte b : hexStrToBytes(hexStr)) {
-                writer.append((char) b);
+    /**
+        \t Insert a tab
+        \b Insert a backspace
+        \n Insert a newline
+        \r Insert a carriage
+        \f Insert a formed
+        \' Insert a single quote character
+        \" Insert a double quote character
+        \\ Insert a backslash
+        \x Insert a hex escape e.g. \x48 represent h
+     */
+    private static int parseHexString(String originStr, StringWriter writer, int offset) throws AnalysisException {
+        if (offset + 4 > originStr.length()) {
+            writer.append(originStr, offset, offset + originStr.length());
+            return originStr.length();
+        }
+        String hexStr = originStr.substring(offset + 2, offset + 4);
```

Comment:
hexStr is definitely two characters, why need to check empty or even length？

Reply:
This is the logic from before, I kept it, and now I think I cant delete it right now